File: frontend-react/src/api/assistant.js
/**
 * AI Writing Assistant API 호출 함수
 * FastAPI 마이크로서비스와 통신
 */
import axios from 'axios'

// 개발 환경에서는 Vite 프록시 사용, 운영에서는 절대 경로
const API_BASE = import.meta.env.VITE_AI_BASE_URL || '/api/assistant'

/**
 * 텍스트 개선
 * @param {string} text - 개선할 텍스트
 * @param {string} tone - 톤 (polite/neutral/friendly/formal/casual)
 * @param {string} language - 언어 (ko/en)
 * @param {string|null} instruction - 추가 지시사항
 * @returns {Promise<Object>} suggestions 배열 포함 응답
 */
export async function improveText({ text, tone = 'polite', language = 'ko', instruction = null }) {
  try {
    const response = await axios.post(`${API_BASE}/improve`, {
      text,
      tone,
      language,
      instruction
    })
    return response.data
  } catch (error) {
    console.error('Improve text error:', error)
    throw new Error(error.response?.data?.detail || '텍스트 개선 중 오류가 발생했습니다')
  }
}

/**
 * 댓글 답변 생성
 * @param {string} comment - 답변할 댓글
 * @param {string|null} context - 추가 컨텍스트
 * @param {string} replyType - 답변 타입 (constructive/apology/explanation/neutral)
 * @param {string} language - 언어 (ko/en)
 * @returns {Promise<Object>} suggestions 배열 포함 응답
 */
export async function generateReply({ comment, context = null, replyType = 'constructive', language = 'ko' }) {
  try {
    const response = await axios.post(`${API_BASE}/reply`, {
      comment,
      context,
      reply_type: replyType,
      language
    })
    return response.data
  } catch (error) {
    console.error('Generate reply error:', error)
    throw new Error(error.response?.data?.detail || '답변 생성 중 오류가 발생했습니다')
  }
}

/**
 * 템플릿 생성
 * @param {string} situation - 상황 (promotion/announcement/apology/explanation)
 * @param {string} topic - 주제/내용
 * @param {string} tone - 톤 (polite/neutral/friendly/formal/casual)
 * @param {string} language - 언어 (ko/en)
 * @returns {Promise<Object>} suggestions 배열 포함 응답
 */
export async function generateTemplate({ situation, topic, tone = 'polite', language = 'ko' }) {
  try {
    const response = await axios.post(`${API_BASE}/template`, {
      situation,
      topic,
      tone,
      language
    })
    return response.data
  } catch (error) {
    console.error('Generate template error:', error)
    throw new Error(error.response?.data?.detail || '템플릿 생성 중 오류가 발생했습니다')
  }
}

File: backend-fastapi/routers/assistant.py
"""
AI Writing Assistant Router
/api/assistant/* 엔드포인트 제공
"""
# 장소영~여기까지: .env 로딩을 최상단에 배치 (서비스 초기화 전에 환경 변수 로드 보장)
import os
from pathlib import Path
from dotenv import load_dotenv

# ✅ routers/assistant.py가 있는 폴더의 상위 폴더(backend-fastapi)의 .env를 로드
env_path = Path(__file__).resolve().parent.parent / ".env"
load_dotenv(dotenv_path=env_path, override=True)

from fastapi import APIRouter, HTTPException, Body
from pydantic import BaseModel, Field
from typing import Optional, List
import logging

import sys

# 프로젝트 루트를 Python 경로에 추가
project_root = Path(__file__).parent.parent
sys.path.insert(0, str(project_root))

from services.writing_assistant import WritingAssistantService, AssistantResponse

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/api/assistant", tags=["assistant"])

# 장소영~여기까지: 서비스 인스턴스 생성 (이 시점에 .env가 이미 로드되어 있어야 함)
assistant_service = WritingAssistantService()


# ==================== Request Models ====================

class ImproveRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000)
    tone: str = Field(default="polite", description="polite/neutral/friendly/formal/casual")
    language: str = Field(default="ko", description="ko/en")
    instruction: Optional[str] = Field(default=None, max_length=500)


class ReplyRequest(BaseModel):
    comment: str = Field(..., min_length=1, max_length=2000)
    context: Optional[str] = Field(default=None, max_length=1000)
    reply_type: str = Field(default="constructive", description="constructive/apology/explanation/neutral")
    language: str = Field(default="ko", description="ko/en")


class TemplateRequest(BaseModel):
    situation: str = Field(..., description="promotion/announcement/apology/explanation")
    topic: str = Field(..., min_length=1, max_length=1000)
    tone: str = Field(default="polite", description="polite/neutral/friendly/formal/casual")
    language: str = Field(default="ko", description="ko/en")


# ==================== Endpoints ====================

@router.post("/improve", response_model=AssistantResponse)
async def improve_text(request: ImproveRequest):
    """
    텍스트 개선
    - 3가지 버전의 개선된 텍스트 제공
    """
    try:
        result = await assistant_service.improve_text(
            text=request.text,
            tone=request.tone,
            language=request.language,
            instruction=request.instruction
        )
        return result
    except Exception as e:
        logger.error(f"Improve text error: {e}")
        raise HTTPException(status_code=500, detail=f"Text improvement failed: {str(e)}")


@router.post("/reply", response_model=AssistantResponse)
async def generate_reply(request: ReplyRequest):
    """
    댓글 답변 생성
    - 3가지 버전의 답변 제공
    """
    try:
        result = await assistant_service.generate_reply(
            comment=request.comment,
            context=request.context,
            reply_type=request.reply_type,
            language=request.language
        )
        return result
    except Exception as e:
        logger.error(f"Generate reply error: {e}")
        raise HTTPException(status_code=500, detail=f"Reply generation failed: {str(e)}")


@router.post("/template", response_model=AssistantResponse)
async def generate_template(request: TemplateRequest):
    """
    템플릿 생성
    - 3가지 버전의 템플릿 제공
    """
    try:
        result = await assistant_service.generate_template(
            situation=request.situation,
            topic=request.topic,
            tone=request.tone,
            language=request.language
        )
        return result
    except Exception as e:
        logger.error(f"Generate template error: {e}")
        raise HTTPException(status_code=500, detail=f"Template generation failed: {str(e)}")


@router.get("/health")
async def health_check():
    """헬스 체크"""
    return {
        "status": "healthy",
        "service": "AI Writing Assistant",
        "api_configured": bool(assistant_service.api_key)
    }
